tests/candev/main.c

```diff
@@ -0,0 +1,213 @@
+/*
+ * Copyright (C) 2019 Nalys
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     tests
+ * @{
+ *
+ * @file
+ * @brief       Test application for the candev abstraction
+ *
+ * @author      Toon Stegen <tstegen@nalys-group.com>
+ * @author      Wouter Symons   <wsymons@nalys-group.com>
+ *
+ * @}
+ */
+
+#define ENABLE_DEBUG (0)
+
+#include <debug.h>
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <isrpipe.h>
+#include "shell.h"
+#include "can/device.h"
+
+#ifdef BOARD_NATIVE
+
+#include <candev_linux.h>
+
+candev_linux_t linux_dev;
+
+#else
+/* add other candev drivers here */
+#endif
+
+#define RX_RINGBUFFER_SIZE 128      //Needs to be a power of 2!
+isrpipe_t rxbuf;
+
+static candev_t *candev = NULL;
+
+static int _send(int argc, char **argv)
+{
+    int ret = 0;
+
+    struct can_frame frame = {
+        .can_id = 1,
+        .can_dlc = 3,
+        .data[0] = 0xAB,
+        .data[1] = 0xCD,
+        .data[2] = 0xEF,
+    };
+
+    if (argc > 1) {
+        if (argc > 1 + CAN_MAX_DLEN) {
+            printf("Could not send. Maximum CAN-bytes: %d\n", CAN_MAX_DLEN);
+            return -1;
+        }
+        for (int i = 1; i < argc; i++) {
+            frame.data[i - 1] = atoi(argv[i]);
+        }
+        frame.can_dlc = argc - 1;
+    }
+
+    ret = candev->driver->send(candev, &frame);
+    if (ret >= 0) {
+        DEBUG("sent using mailbox: %d\n", ret);
+    }
+    else {
+        puts("Failed to send CAN-message!");
+    }
```

Comment:
I would reverse the logic here

```suggestion
    if (ret < 0) {
        puts("Failed to send CAN-message!");
    }
    else {
        DEBUG("sent using mailbox: %d\n", ret);
    }
```